packages/tic-tac-toe-bot/src/bot.ts

```diff
@@ -0,0 +1,184 @@
+import { ethers } from "ethers";
+import { v4 } from "uuid";
+
+const bn = ethers.utils.bigNumberify;
+const ZERO = bn(0);
+
+function checkDraw(board) {
+  return board.every(row => row.every(square => !bn(square).eq(ZERO)));
+}
+
+function checkVictory(board, player) {
+  return (
+    checkHorizontalVictory(board, player) ||
+    checkVerticalVictory(board, player) ||
+    checkDiagonalVictory(board, player) ||
+    checkCrossDiagonalVictory(board, player)
+  );
+}
+
+function checkHorizontalVictory(board, player) {
+  let idx;
+  const victory = board.some((row, index) => {
+    idx = index;
+    return row.every(square => bn(square).eq(bn(player)));
+  });
+
+  if (victory) {
+    return {
+      idx,
+      winClaimType: 0
+    };
+  }
+  return false;
+}
+
+function checkVerticalVictory(board, player) {
+  let idx;
+  const victory = board[0].some((columnStart, index) => {
+    idx = index;
+    return (
+      bn(columnStart).eq(bn(player)) &&
+      bn(board[1][index]).eq(bn(player)) &&
+      bn(board[2][index]).eq(bn(player))
+    );
+  });
+
+  if (victory) {
+    return {
+      idx,
+      winClaimType: 1
+    };
+  }
+  return false;
+}
+
+function checkDiagonalVictory(board, player) {
+  const victory =
+    bn(board[0][0]).eq(bn(player)) &&
+    bn(board[1][1]).eq(bn(player)) &&
+    bn(board[2][2]).eq(bn(player));
+
+  if (victory) {
+    return {
+      idx: 0,
+      winClaimType: 2
+    };
+  }
+  return false;
+}
+
+function checkCrossDiagonalVictory(board, player) {
+  const victory =
+    bn(board[0][2]).eq(bn(player)) &&
+    bn(board[1][1]).eq(bn(player)) &&
+    bn(board[2][0]).eq(bn(player));
+
+  if (victory) {
+    return {
+      idx: 0,
+      winClaimType: 3
+    };
+  }
+  return false;
+}
+
+function respond(node, ethAddress, { data: { appInstanceId, newState } }) {
+  const winner = ethers.utils.bigNumberify(newState[2]).toNumber();
+
+  if (winner === 0) {
```

Comment:
Should define `enums` for these numbers. I think the contract already has labels for them.